client/ResponseUnpacker.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <stdexcept>
+#include <iomanip>
+#include <sstream>
+#include <cstdint>
+
+class ResponsePacketUnpacker {
+public:
+    explicit ResponsePacketUnpacker(const std::vector<uint8_t>& data);
+
+    std::vector<std::string> unpack();
+
+private:
+    struct Packet {
+        uint8_t version;
+        uint16_t code;
+        uint32_t payload_size;
+        std::vector<uint8_t> payload;
+    };
+
+    Packet packet;
+    const std::vector<uint8_t>& data;
+
+    void unpack_general();
+
+    std::string unpack_uuid(const std::vector<uint8_t>& payload);
+
+    void unpack_aes_key(std::vector<std::string>& result);
+
+    void unpack_file_received(std::vector<std::string>& result);
+};
+
+uint32_t getCode(std::shared_ptr<std::vector<std::string>> data);
+
+std::string getAESKey(std::shared_ptr<std::vector<std::string>> packet);
+
+uint32_t getCRC(std::shared_ptr<std::vector<std::string>> packet);

```

Comment:
why not pass a non-const-reference?
also, these are exactly why the polymorphism should be used here. these functions make everything implementation dependent and should probably not be static and fitting for every packet (easy failure).

make an abstract (ish) packet interface with specializations